### PR TITLE
api: check status code instead of body

### PR DESCRIPTION
**Pull request by rleungx**
See pingcap/pd#2319.

### CODE REVIEW

File: pkg/api/pdapi.go [hunk 1]
```diff
@@ -416,16 +416,8 @@ func (pc *PDClient) RemoveStoreEvict(host string) error {
...
-		if err != nil {
-			// TODO: also check HTTP status code
-			if bytes.Contains(body, []byte("scheduler not found")) {
-				log.Debugf("Store leader evicting scheduler does not exist, ignore.")
-				return nil
-			}
-			return err
-		}
-		return nil
+		_, err := pc.httpClient.Delete(endpoint, nil)
```
Comment (AstroProfundis) [inline pkg/api/pdapi.go:419]: We should change this to ignore 404 code.

Comment (AstroProfundis) [reply pkg/api/pdapi.go:419]: And the workaround should also be kept for compatibility with older PD versions.

Comment (rleungx) [reply pkg/api/pdapi.go:419]: Oh, Does the other APIs use `Delete` also need to ignore the 404 error?

Comment (lonng) [reply pkg/api/pdapi.go:419]: It's better to check the other using APIs.

**Pull request merged**